src/mpid/ch4/netmod/ofi/ofi_pre.h

```diff
@@ -284,7 +284,7 @@ typedef struct {
 } MPIDI_OFI_win_t;
 
 /* Maximum number of network interfaces CH4 can support. */
-#define MPIDI_OFI_MAX_NICS 8
+#define MPIDI_OFI_MAX_NICS 128
```

Comment:
I think we can remove this macro and rely on `MPIR_CVAR_CH4_OFI_MAX_NICS` to control the number of active NICs to use. I think we should reset `MPIR_CVAR_CH4_OFI_MAX_NICS` default to `8`.

Reply:
@hzhou Cool this is better. I can give it a go by next week.

I would also like second opinions from Aurora folks since they introduced the limit. Do you know a good contact?

Reply:
cc @zhenggb72 @sagarth

Reply:
@zhenggb72 @sagarth Some background on this PR. The 8 NIC limit is a blocker for AWS customers to use newer platforms with e.g. 32 NICs. We are exploring the possibility to raise or preferably remove the limit.

Do you see any downside of doing so?

Reply:
I do like Hui's suggestion. We also went to keep the ability to change the number of NICs with a CVAR, in case we only want to use the NICs on first socket. Its an easier control.

@wenduwan Maybe you could set a machine specific value of this CVAR on your platforms?

Reply:
@hzhou After reading the usage of `MPIDI_OFI_MAX_NICS` closer, I think it requires a larger change to eliminate it.

I'm especially curious about these static counters https://github.com/pmodels/mpich/blob/69522c242b8106c1fa0dfa7fa52944eaadafbdde/src/mpid/ch4/netmod/ofi/globals.c#L10-L19

Do you think we can/should make them dynamically sized instead?

Reply:
Yes, I think they can be dynamically allocated

Reply:
@tarudoodi Apologies I missed your comment last week. I have retained the CVAR after the change. Please let me know if that is sufficient for your purpose.